backend/src/despesas/despesas.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable, BadRequestException, NotFoundException } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { CreateDespesaDto } from './dto/create-despesa.dto';
+import { UpdateDespesaDto } from './dto/update-despesa.dto';
+
+@Injectable()
+export class DespesasService {
+  constructor(private prisma: PrismaService) {}
+
+  async findAll() {
+    return this.prisma.despesa.findMany({
+      orderBy: { data: 'desc' },
+    });
+  }
+
+  async findOne(id: string) {
+    const despesa = await this.prisma.despesa.findUnique({
+      where: { id },
+    });
+
+    if (!despesa) {
+      throw new NotFoundException(`Despesa com ID ${id} não encontrada`);
+    }
+
+    return despesa;
+  }
+
+  async create(createDespesaDto: CreateDespesaDto) {
+    const dataDespesa = new Date(createDespesaDto.data);
+
+    if (dataDespesa > new Date()) {
+      throw new BadRequestException('Não é permitido informar uma data futura.');
+    }
+
+    try {
+      const despesa = await this.prisma.despesa.create({
+        data: {
+          descricao: createDespesaDto.descricao,
+          data: dataDespesa,
+          valor: createDespesaDto.valor,
+          fornecedor: createDespesaDto.fornecedor,
+          observacoes: createDespesaDto.observacoes,
+        },
+      });
+
+      return {
+        message: 'Sucesso! A despesa foi incluída com sucesso.',
+        data: despesa,
+      };
+    } catch (error) {
+      if (error.code === 'P2002') {
+        throw new BadRequestException(
+          'Não foi possível salvar o registro! Já existe uma despesa com a mesma descrição.'
+        );
+      }
+      throw error;
+    }
+  }
+
+  async update(id: string, updateDespesaDto: UpdateDespesaDto) {
```

Comment:
o update ta sempre retornando sucesso mesmo que a gente envie um objeto com propriedades que não existem

o ideal seria validar os dados se podem ser recebidos. e a validação do campo de data não pode fazer update pra undefined, que se caso ela nao for enviada no objeto de edição, a data original vai se perder